parsers/ES.py

```diff
@@ -112,18 +157,44 @@ def check_valid_parameters(
         )
 
 
-def fetch_island_data(
-    zone_key: ZoneKey, session: Session, target_datetime: datetime | None
-) -> list[Response]:
-    """Fetch data for the given zone key."""
+def fetch_data_ree(
+    zone_key: ZoneKey, session: Session, target_datetime: Optional[datetime], tz: str
+) -> Dict:
     if target_datetime is None:
-        date = target_datetime
+        date = datetime.now(tz=ZoneInfo(tz)).strftime("%Y-%m-%d")
     else:
         date = target_datetime.strftime("%Y-%m-%d")
+    system = ZONE_MAPPING[zone_key]
+    zone = API_CODE_MAPPING[zone_key]
```

Comment:
As specified before I think it would be better to combine these lists into one.

Something like:
```python
{
  ZoneKey("ES"): {
    "API_CODE": "code",
    "SYSTEM": "system",
    "TZ": "timezone"
  },
  # ...
}
```
It would remove a lot of duplicate code and have a nice and clear object grouping for each zone.